hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2848,3 +2856,213 @@ func reconcileClusterPrometheusRBAC(ctx context.Context, c client.Client, namesp
 
 	return nil
 }
+
+func (r *HostedClusterReconciler) reconcileMachineApprover(ctx context.Context, hcluster *hyperv1.HostedCluster, hcp *hyperv1.HostedControlPlane) error {
+	controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name)
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(controlPlaneNamespace), controlPlaneNamespace); err != nil {
+		return fmt.Errorf("failed to get control plane namespace: %w", err)
+	}
+
+	// Reconcile machine-approver role
+	role := machineapprover.Role(controlPlaneNamespace.Name)
+	if _, err := r.CreateOrUpdate(ctx, r.Client, role, func() error {
+		return reconcileMachineApproverRole(role)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver role: %w", err)
+	}
+
+	// Reconcile machine-approver service account
+	sa := machineapprover.ServiceAccount(controlPlaneNamespace.Name)
+	if _, err := r.CreateOrUpdate(ctx, r.Client, sa, NoopReconcile); err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver service account: %w", err)
+	}
```

Comment:
I was expecting the pattern with logging and tracing for the operation output, but we can do that later in a different PR to consolidate across the controller.